Core/Core/Files/View/FileList/FileListViewController.swift

```diff
@@ -453,6 +467,28 @@ extension FileListViewController: UITableViewDataSource, UITableViewDelegate {
         configuration.performsFirstActionWithFullSwipe = false
         return configuration
     }
+
+    private func isItemAvailableOffline(fileID: String?) -> Bool {
```

Comment:
This seems to be a good candidate to be moved to `OfflineFileInteractor`. This viewcontroller could also depend on this `OfflineFileInteractor` instead of `OfflineModeInteractor`.

Reply:
good insight, refactored.